tests/v2/validation/upgrade/cloud_provider_aws_migration.go

```diff
@@ -0,0 +1,290 @@
+package upgrade
+
+import (
+	"os"
+	"testing"
+
+	provv1 "github.com/rancher/rancher/pkg/apis/provisioning.cattle.io/v1"
+	v1 "github.com/rancher/rancher/pkg/apis/rke.cattle.io/v1"
+	"github.com/rancher/rancher/tests/v2/validation/provisioning/permutations"
+	"github.com/rancher/shepherd/clients/rancher"
+	management "github.com/rancher/shepherd/clients/rancher/generated/management/v3"
+	steveV1 "github.com/rancher/shepherd/clients/rancher/v1"
+	"github.com/rancher/shepherd/extensions/clusters"
+	"github.com/rancher/shepherd/extensions/kubectl"
+	"github.com/rancher/shepherd/extensions/services"
+	"github.com/rancher/shepherd/extensions/workloads"
+	"github.com/rancher/shepherd/extensions/workloads/pods"
+	namegen "github.com/rancher/shepherd/pkg/namegenerator"
+	"github.com/sirupsen/logrus"
+	"github.com/stretchr/testify/require"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	rancherShellSettingID       = "shell-image"
+	controlPlaneMatchLabelKey   = "rke.cattle.io/control-plane-role"
+	kubeControllerManagerArgKey = "kube-controller-manager-arg"
+	enableLeaderMigration       = "enable-leader-migration"
+	fleetNamespace              = "fleet-default"
+
+	outOfTreeAWSYamlPath = "../provisioning/resources/out-of-tree/aws.yml"
+)
+
+var (
+	group int64
+	user  int64
+)
+
+// runKubectlCommand is a helper that takes a client of a cluster and a command, then creates all needed
+// resources in order to run the command on the cluster via a corev1.job
+func runKubectlCommand(client *rancher.Client, cmd, v3ClusterName string) error {
+	jobName := namegen.AppendRandomString("job")
+	imageSetting, err := client.Management.Setting.ByID(rancherShellSettingID)
+	if err != nil {
+		return err
+	}
+
+	jobTemplate := workloads.NewJobTemplate(jobName, kubectl.Namespace)
+	args := []string{
+		cmd,
+	}
+	command := []string{"/bin/sh", "-c"}
+	securityContext := &corev1.SecurityContext{
+		RunAsUser:  &user,
+		RunAsGroup: &group,
+	}
+	volumeMount := []corev1.VolumeMount{
+		{Name: "config", MountPath: "/root/.kube/"},
+	}
+	container := workloads.NewContainer(
+		jobName, imageSetting.Value, corev1.PullAlways, volumeMount, nil, command, securityContext, args)
+	jobTemplate.Spec.Template.Spec.Containers = append(jobTemplate.Spec.Template.Spec.Containers, container)
+
+	return kubectl.CreateJobAndRunKubectlCommands(v3ClusterName, jobName, jobTemplate, client)
+}
+
+// enableLeaderMigrationInMachineSelector adds the enable-leader-migration flag to the kube-controller-manager-arg-key
+// in the provided []RKESystemConfig in order to enable leader migration
+// returns the updated []RKESystemConfig
```

Comment:
nit; can this be just on the second line? It shouldn't overcrowd it and eliminates just another comment line.

Reply:
sure